File: common/mrtjp/projectred/compatibility/Services.java
package mrtjp.projectred.compatibility;

import mrtjp.projectred.compatibility.tconstruct.ProxyTConstruct;
import mrtjp.projectred.compatibility.thermalexpansion.ProxyThermalExpansion;
import cpw.mods.fml.common.Loader;

public class Services
{
    /** loaded interactions **/
    public static boolean loadTConstruct = false;
    public static boolean loadTExpansion = false;

    public static void loadServices()
    {
        if (Loader.isModLoaded("TConstruct"))
        {
            loadTConstruct = true;
            tcProxy = new ProxyTConstruct();
        }
        
        if (Loader.isModLoaded("ThermalExpansion"))
        {
            loadTExpansion = true;
            teProxy = new ProxyThermalExpansion();
        }
    }

    private static ProxyTConstruct tcProxy;
    public static ProxyTConstruct getTCProxy()
    {
        return tcProxy;
    }
    
    private static ProxyThermalExpansion teProxy;
    public static ProxyThermalExpansion getTEProxy()
    {
        return teProxy;
    }
}

File: common/mrtjp/projectred/compatibility/Services.scala
package mrtjp.projectred.compatibility

import cpw.mods.fml.common.Loader
import mrtjp.projectred.compatibility.tconstruct.ProxyTConstruct
import mrtjp.projectred.compatibility.thermalexpansion.ProxyThermalExpansion
import mrtjp.projectred.compatibility.treecapitator.ProxyTreecapitator

object Services
{
    /** loaded interactions **/
    var loadTConstruct = false
    var loadTExpansion = false
    var loadTreecapitator = false

    def loadServices()
    {
        if (Loader.isModLoaded("TConstruct"))
        {
            loadTConstruct = true
            tcProxy = new ProxyTConstruct
        }
        if (Loader.isModLoaded("ThermalExpansion"))
        {
            loadTExpansion = true
            teProxy = new ProxyThermalExpansion
        }
        if (Loader.isModLoaded("TreeCapitator") && Loader.isModLoaded("ProjRed|Exploration"))
        {
            loadTreecapitator = true
            treecapProxy = new ProxyTreecapitator
        }
    }

    var tcProxy:ProxyTConstruct = null
    var teProxy:ProxyThermalExpansion = null
    var treecapProxy:ProxyTreecapitator = null
}